dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/subsystem/SubsystemFeatureDefinitionImpl.java

```diff
@@ -484,6 +484,33 @@ public String toString() {
         return icons;
     }
 
+    /*
+     * (non-Javadoc)
+     *
+     * @see com.ibm.ws.kernel.feature.provisioning.ProvisioningFeatureDefinition#getEndpointIcons()
+     */
+    @Override
+    public Collection<String> getEndpointIcons() {
+        Collection<String> icons = new ArrayList<String>();
+        String endpointIconsHeader = getHeader("Subsystem-Endpoint-Icons");
+        if (endpointIconsHeader != null) {
+            String[] endpointIcons = endpointIconsHeader.split(",");
+            String endpointIconCategory = "";
+            for (String endpointIcon : endpointIcons) {
+                String[] endpointIconDirectives = endpointIcon.split(";");
+                if (endpointIconDirectives.length <= 1 && endpointIcon.indexOf("=") >= 0) {
+                    String[] endpointIconSplit = endpointIcon.split("=");
+                    endpointIconCategory = endpointIconSplit[0].trim();
+                    icons.add(endpointIconCategory + "/" + endpointIconSplit[1].trim());
+                    continue;
+                }
+                // Assume the first part is always the file URL
+                icons.add(endpointIconCategory + "/" + endpointIconDirectives[0].trim());
```

Comment:
The example I see for Subsystem-Endpoint-Icons is this:
```
Subsystem-Endpoint-Icons: \
 clientManagement=OSGI-INF/clientManagement_142.png,
   OSGI-INF/clientManagement_78.png;size=78,
   OSGI-INF/clientManagement_142.png;size=142
```
So the category ends up being `clientManagement`.  We then add the path based of the category.  Will `clientManagement/OSGI-INF/clientManagement_142.png` be added twice to the result?

Reply:
Yes.  The redundancy is similar to what we see when processing the Subsystem-Icon header.